sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -32,7 +45,23 @@ export class EducationProgramOfferingController {
     @Body() payload: SaveEducationProgramOfferingDto,
     @Param("locationId") locationId: number,
     @Param("programId") programId: number,
+    @UserToken() userToken: IInstitutionUserToken,
   ): Promise<number> {
+    const requestedLoc = await this.locationService.getInstitutionLocationById(
+      locationId,
+    );
+    if (
+      userToken.authorizations.institutionId !== requestedLoc.institution.id
```

Comment:
The fixes in this way look good to complete this ticket, but I believe that we should move this check inside the InstitutionLocationGuard. What is happening now is that we are not checking the locations for admins.
To fix it in the InstitutionLocationGuard, I believe that we should.

1. Load the locations Ids when the user is an Admin on JwtStrategy
2. Remove the below code from InstitutionLocationGuard and let the location ids be validated for Admin also.
```
if (authorizations.isAdmin()) {
   return true;
}
```